crates/ruff_cli/src/lib.rs

```diff
@@ -386,21 +386,24 @@ pub fn check(args: CheckCommand, log_level: LogLevel) -> Result<ExitStatus> {
                         fix_mode,
                         unsafe_fixes,
                     )?;
-                    printer.write_continuously(&mut writer, &messages)?;
+                    printer.write_continuously(&mut writer, &diagnostics)?;
                 }
                 Err(err) => return Err(err.into()),
             }
         }
     } else {
         // Generate lint violations.
-        let diagnostics = if is_stdin {
-            commands::check_stdin::check_stdin(
-                cli.stdin_filename.map(fs::normalize_path).as_deref(),
-                &pyproject_config,
-                &overrides,
-                noqa.into(),
-                fix_mode,
-            )?
+        let (diagnostics, checked_files) = if is_stdin {
+            (
+                commands::check_stdin::check_stdin(
+                    cli.stdin_filename.map(fs::normalize_path).as_deref(),
+                    &pyproject_config,
+                    &overrides,
+                    noqa.into(),
+                    fix_mode,
+                )?,
+                1,
```

Comment:
Flagging here that I'm returning `1` here for the number of affected files if we're reading from `stdin`.

Please let me know if this is okay!